tests/test_consumer/test_extractors/test_extractors_dataset.py

```diff
@@ -38,6 +61,83 @@ def test_extractors_extract_dataset_hdfs():
     assert symlinks == []
 
 
+def test_extractors_extract_dataset_hdfs_with_patition():
+    dataset = OpenLineageDataset(
+        namespace="hdfs://test-hadoop:9820",
+        name="/user/hive/warehouse/mydb.db/mytable/business_dt=2025-01-01/reg_id=99/part_dt=2025-01-01",
+    )
+
+    dataset, symlinks = extract_dataset_and_symlinks(dataset)
+
+    assert dataset == DatasetDTO(
+        location=LocationDTO(
+            type="hdfs",
+            name="test-hadoop:9820",
+            addresses={"hdfs://test-hadoop:9820"},
+        ),
+        name="/user/hive/warehouse/mydb.db/mytable",
+    )
+    assert symlinks == []
+
+
+def test_extractors_extract_batch_dataset_hdfs():
+    """
+    There is two datasets name in event. They should be union into one, excluding partition.
+    """
+    event_time = datetime(2024, 7, 5, 9, 7, 9, 849000, tzinfo=timezone.utc)
+    run_id = UUID("01908224-8410-79a2-8de6-a769ad6944c9")
+    operation_id = UUID("01908225-1fd7-746b-910c-70d24f2898b1")
+    event = OpenLineageRunEvent(
+        eventType=OpenLineageRunEventType.RUNNING,
+        eventTime=event_time,
+        job=OpenLineageJob(
+            namespace="local://some.host.com",
+            name="mysession.execute_some_command",
+            facets=OpenLineageJobFacets(
+                jobType=OpenLineageJobTypeJobFacet(
+                    jobType=OpenLineageJobType.JOB,
+                    processingType=OpenLineageJobProcessingType.BATCH,
+                    integration=OpenLineageJobIntegrationType.SPARK,
+                ),
+            ),
+        ),
+        run=OpenLineageRun(
+            runId=operation_id,
+            facets=OpenLineageRunFacets(
+                parent=OpenLineageParentRunFacet(
+                    job=OpenLineageParentJob(
+                        namespace="local://some.host.com",
+                        name="mysession",
+                    ),
+                    run=OpenLineageParentRun(
+                        runId=run_id,
+                    ),
+                ),
+            ),
+        ),
+        outputs=[
+            OpenLineageOutputDataset(
+                namespace="hdfs://test-hadoop:9820",
+                name="/user/hive/warehouse/mydb.db/mytable/business_dt=2025-01-01/reg_id=99/part_dt=2025-01-01",
+            ),
+            OpenLineageOutputDataset(
+                namespace="hdfs://test-hadoop:9820",
+                name="/user/hive/warehouse/mydb.db/mytable/business_dt=2025-02-01/reg_id=99/part_dt=2025-01-01",
+            ),
+        ],
+    )
+
+    extracted = extract_batch([event])
```

Comment:
Tests for batch extraction are in another file

Reply:
Hmm, but this test mostly for testing `extract_batch` like for column_lineage we also has test `extract_batch` in `test_extractors_column_lineage.py`. I'm thinking that create separate file for one test is overhead. And add this into `_batch_spark.py` also strange.

Reply:
Ok, but this looks strange for me

Reply:
move it to `test_extractors_batch_spark.py`